### DISCUSSION TITLE
为什么在有的lookup 中需要考虑default 值，有些却不考虑，实践中应该遵循什么规则？

### DISCUSSION THREAD

**Discussion by readygo67** [z2o-k7e/halo2-step-by-step#50] [Q&A]
在 https://github.com/zkp-co-learning/halo2-step-by-step/blob/main/halo2-tutorials/src/chap_4/circuit_3.rs 中考虑default 值，

![image](https://github.com/zkp-co-learning/halo2-step-by-step/assets/78890754/e55044c0-fe29-4543-8239-8059e206d4ae)

而在 https://github.com/zkp-co-learning/halo2-step-by-step/blob/main/halo2-tutorials/src/chap_4/circuit_2.rs 中却不需要考虑。

**Comment by qizhou** [comment]
因为circuit_3里面，selector = 0时， (b = 0, v = 0)在lookup_table中间没有对应的值，所以需要处理default case。

在circuit_2里面，selector = 0的时候, lookup_value = 0，lookup_table有对应的值，所以不需要 default case。
